reader.go

```diff
@@ -85,6 +90,18 @@ func (this *Reader) Read(p []byte) (n int, err error) {
 
 	// if we experienced an iconv error, check it
 	if err != nil {
+		// EINVAL:
+		// An incomplete multibyte sequence is encountered in the input,
+		// and the input byte sequence terminates after it.
+		if err == syscall.EINVAL {
```

Comment:
instead of reading more data, can we use similar logic to E2BIG? Meaning, can't we just ignore EINVAL as long as we've written at least 1 byte during conversion?

`if err != nil {
	switch (err) {
	case syscall.EINVAL:
		// EINVAL can happen if the end of our buffer contains an
		// incomplete multibyte sequence. Ignore unless we know its
		// at the beginning of the input buffer.
		if bytesRead == 0 {
			this.err = err
		}
	case syscall.E2BIG:
		// E2BIG errors can be ignored (we'll get them often) as long
		// as at least 1 byte was written. If we experienced an E2BIG
		// and no bytes were written then the buffer is too small for
		// even the next character
		if bytesWritten == 0 {
			this.err = err
		}
	default:
		// track anything else
		this.err = err
	}
}`